packages/@aws-cdk/aws-lambda/lib/languages/lambda-builder.ts

```diff
@@ -0,0 +1,106 @@
+import assets = require('@aws-cdk/assets');
+import { BuildAsset } from '@aws-cdk/assets';
+import cdk = require('@aws-cdk/cdk');
+import { join } from 'path';
+import { Code } from '../code';
+import { CfnFunction } from '../lambda.generated';
+import { Runtime } from '../runtime';
+
+interface LambdaBuilderRequest {
+  jsonrpc: string;
+  method: string;
+  id: number;
+  params: {
+    __protocol_version: string;
+    capability: {
+      language: string;
+      dependency_manager: string;
+      application_framework: string | null;
+    },
+    source_dir: string;
+    artifacts_dir: string;
+    scratch_dir: string;
+    manifest_path: string;
+    runtime: string;
+    optimizations: object;
+    options: object;
+  }
+}
+
+export interface LambdaBuilderCodeProps {
+  path: string;
+  runtime: Runtime;
+  language: string;
+  dependencyManager: string;
+  manifestName: string;
+}
+
+export class LambdaBuilderCode extends Code {
+  public readonly isInline: boolean = false;
+
+  protected readonly path: string;
+  protected readonly runtime: Runtime;
+
+  protected readonly language: string;
+  protected readonly dependencyManager: string;
+  protected readonly manifestName: string;
+
+  protected asset?: BuildAsset;
+
+  constructor(props: LambdaBuilderCodeProps) {
+    super();
+    this.path = props.path;
+    this.runtime = props.runtime;
+    this.language = props.language;
+    this.dependencyManager = props.dependencyManager;
+    this.manifestName = props.manifestName;
+  }
+
+  public bind(construct: cdk.Construct) {
+    // If the same AssetCode is used multiple times, retain only the first instantiation.
+    if (!this.asset) {
+      const mount = '/tmp/cdk.out/';
```

Comment:
Is this the directory that will mounted into the container? Do you need to handle windows paths here?

Be careful with `/tmp` directories on the host. These are not shared by default with docker and has caused some pain with SAM CLI customers (especially on Windows). 

If you are bind mounting in 'ro' mode (like SAM CLI does), you will need to be aware that Go likes to write files into the source directory on the container. There is a change I am working on to copy code instead of bind mount to get around these. We are looking at this as an alternative/fall back for when bind mounts fail. We have seen the mount fail on Windows when using VPNs. Something to keep in mind while making design decisions here.